### PR TITLE
fix(refactor): Fix duplicate code for issue #1265

### PR DESCRIPTION
**Pull request by mayank-pathakk**
This PR aims to solve the duplicate code issue in #1265 .

### CODE REVIEW

File: src/emit/substrate/target.rs [hunk 1]
```diff
@@ -526,15 +532,8 @@ impl<'a> TargetRuntime<'a> for SubstrateTarget {
...
-        let offset = unsafe {
-            binary.builder.build_gep(
-                binary.context.i8_type().array_type(SCRATCH_SIZE),
-                binary.scratch.unwrap().as_pointer_value(),
-                &[i32_zero!(), length],
-                "data_offset",
-            )
-        };
-
+        let offset = create_gep_instruction!(binary, binary.scratch.unwrap(), length);
```
Comment (LucasSte) [inline src/emit/substrate/target.rs:535]: The name should be specific to what this function/macro is doing. `create_gep_instruction` is too generic. What you are doing here is fetching a pointer to an offset of the scratch buffer.

Comment (mayank-pathakk) [reply src/emit/substrate/target.rs:535]: Okay. I'll amend this and keep it in mind for the remaining of the refactoring

File: src/emit/substrate/target.rs [hunk 2]
```diff
@@ -19,6 +19,19 @@ use inkwell::{AddressSpace, IntPredicate};
 use solang_parser::pt::Loc;
 use std::collections::HashMap;
 
+macro_rules! create_gep_instruction {
```
[the remark this answers is not in the document]
Comment (xermicus) [reply src/emit/substrate/target.rs:22]: That's not what macros are for. If you'd like to learn more about macros, I can recommend the [macro section in the rust book](https://doc.rust-lang.org/book/ch19-06-macros.html#the-difference-between-macros-and-functions).

**Comment by Nirhar** [comment]
I'd like to work on this issue, if its still open!

**Pull request closed without merging**